### DISCUSSION TITLE
How can I overwrite default keybind to toggle the terminal ?

### DISCUSSION THREAD

**Discussion by Raekker** [folke/snacks.nvim#63] [Q&A] [answered]
As in the title. I've tried couple of things based on [docs for overwriting keybinds](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps), however it doesn't seem to do the trick. So I've tried simply overwriting keybind - didn't work, then disable keybind and overwrite - also didn't work, even just disabling keybind doesn't work, unless I'm doing something incorrectly. To give it some more context, I want to have the same keybind for toggling the terminal, however I want it to spawn with specified shell and not default one.
My config is as follows:

```lua 
 {
    "folke/snacks.nvim",
    priority = 1000,
    lazy = false,
    ---@type snacks.Config
    opts = {
      terminal = {
        win = {
          border = "rounded",
          position = "float",
        },
      },
    },
    keys = {
      { "<c-/>", false },
      {
        "<c-/>",
        function()
          Snacks.terminal.toggle("nu")
        end,
        desc = "Toggle Terminal",
      },
    },
  },
```

**Comment by f1uff3h** [comment, marked as answer]
setting ` { "<c-/>", false }` disables your keymap so second one is not registered. simply delete this line and it should work for you. see screenshots below
![image](https://github.com/user-attachments/assets/c72aac51-2f5b-40b0-9f19-414c63e09c8c)
![image](https://github.com/user-attachments/assets/5784f851-dc45-4157-8b4d-b9ec7adf0ebf)

but if you just want to start your terminal with Nushell you're better off setting `LazyVim.terminal.setup("nu")` in your options.lua (as per LazyVim docs)

**Reply by Raekker** [reply]
Thank you so much for help
1. Setting shell via options worked
2. Removing the first line disabling keybind and keeping the one to overwrite doesn't work for whatever reason

I got what I wanted though, so I guess that an issue for later
![Kooha-2024-11-12-12-09-09](https://github.com/user-attachments/assets/01b6693b-2bc8-4436-a1a1-6e278238fb9b)

**Reply by f1uff3h** [reply]
I was actually wrong commenting `{ "<c-/>", false }` out doesn't help as the configuration works with the line set as well. I spun up another container and it definitely works with setting Snacks.terminal.toggle("nu") so you might want to check if you override the setting in keymaps or somewhere else. See below video. Anyway I think setting shell via LazyVim.terminal.setup is preffered, unless you want different shells and terminal configurations on different shortcuts

https://github.com/user-attachments/assets/b2e137f1-6eea-48f4-a642-ebc53136fbda
